plugins/eval.lua

```diff
@@ -29,4 +31,23 @@ command.add("core.docview", {
       end
     })
   end,
+
+  ["eval:selected"] = function(dv)
+    if dv.doc:has_selection() then
+      local text = dv.doc:get_text(dv.doc:get_selection())
+      dv.doc:text_input(eval(text))
+    else
+      local line = dv.doc:get_selection()
+      local text = dv.doc.lines[line]
+      dv.doc:insert(line+1, 0, "= " .. eval(text) .. "\n")
+    end
+  end,
 })
+
+
+contextmenu:register("core.docview", {
+  { text = "Evaluate Selected",  command = "eval:selected" }
+})
+
+
+keymap.add { ["ctrl+return"] = "eval:selected" }
```

Comment:
I was going to merge this but noticed that `ctrl+return` is already binded in core to insert empty line after current line